trio/tests/test_ssl.py

```diff
@@ -1109,6 +1109,76 @@ def close_hook():
             await client.do_handshake()
 
 
+async def test_selected_alpn_protocol_before_handshake():
+    client, server = ssl_memory_stream_pair()
+
+    with pytest.raises(NoHandshakeError):
+        client.selected_alpn_protocol()
+        server.selected_alpn_protocol()
```

Comment:
In order to make the test actually cover both lines, you need to have two separate `pytest.raises()` blocks. Otherwise what happens is that either the block exits at the first line because it raises (and it never executes the second one); or the first line doesn't raise, but you won't see that because the second one does raise.

```suggestion

    with pytest.raises(NoHandshakeError):
        server.selected_alpn_protocol()
```

The same applies to the other similar tests. This is the reason why codecov is complaining, and you fix it (assuming the tests pass) you will get full coverage again :)

Reply:
I see, I'll fix that. Thanks for your help with that!